fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordContextProperties.java

```diff
@@ -61,6 +61,11 @@ public final class LuceneRecordContextProperties {
      */
     public static final RecordLayerPropertyKey<Double> LUCENE_MERGE_MAX_SIZE = RecordLayerPropertyKey.doublePropertyKey("com.apple.foundationdb.record.lucene.mergeMaxSize", 5.0);
 
+    /**
+     * Count of segments after which to merge for ordinary full-text search with Lucene.
```

Comment:
Is limiting this a requirement of the new index? Or is this just needed because merging has additional cost, and we could more easily run into transaction limits?

Reply:
I suspect that this is just to facilitate testing, and have better control over how many merges happen.

Reply:
I added it for testing, though I could imagine that the work going on in parallel to do merges out-of-band could want to control this. It is not necessary with the new index and defaults to what the Lucene default would be.